src/fhirpath/search.py

```diff
@@ -731,59 +724,72 @@ def rev_include(self, main_query_result: EngineResult) -> List[QueryResult]:
 
         return include_queries
 
-    def add_term(self, normalized_data, terms_container):
-        """ """
+    def add_terms(self, normalized_data: Tuple, terms_container: List):
```

Comment:
If I'm not mistaking, this is not recursive anymore, can you init the terms_container inside this function and return it instead ? maybe I'm missing something though.